src/plugins/core/filters.ts

```diff
@@ -109,7 +110,15 @@ export class FiltersPlugin extends CorePlugin<FiltersState> implements FiltersSt
         this.onDeleteColumnsRows(cmd);
         break;
       case "CREATE_FILTER_TABLE": {
-        const zone = union(...cmd.target);
+        let zone = union(...cmd.target);
+        const sheetId = cmd.sheetId;
+        if (zone.top == zone.bottom && zone.left == zone.right) {
+          [zone] = adjecentCellsSelection(
+            sheetId,
+            { sheetId: sheetId, row: zone.top, col: zone.right },
+            this.getters.getCell
+          );
+        }
```

Comment:
Do it upstream. It's not the role of the plugin to extend the zone. It should be done in the action which trigger this command.